qiskit/circuit/library/standard_gates/u1.py

```diff
@@ -127,7 +128,7 @@ def control(
             ControlledGate: controlled version of this gate.
         """
         if num_ctrl_qubits == 1:
-            gate = CU1Gate(self.params[0], label=label, ctrl_state=ctrl_state)
+            gate: Gate = CU1Gate(self.params[0], label=label, ctrl_state=ctrl_state)
```

Comment:
Since `CU1Gate` is clearly a `Gate` object, is this type hint necessary?

Reply:
Otherwise it is assigned `CU1Gate` type, which is more narrow

Reply:
Hmm but why is that an issue? In the elif case below, the explicit `Gate` typehint for `MCU1Gate` also doesn't seem to be required (or in other places where gates are created) 🤔